### ISSUE TITLE
Stats: Referrers: Not Possible to Mark a Referrer as Spam

### ISSUE THREAD

**Issue by thehenrybyrd** [wordpress-mobile/WordPress-iOS#14055]
In Calypso, referrers which are not whitelisted can be marked spam by tapping the three dots to the right of the referrer. This is not the case in the WordPress iOS app (or Android).

Expected:
I expected to be able to mark referrers as spam from within the app.

Actual: 
The option to mark referrers as spam doesn’t exist within the app.

Stats referrers in WPiOS and Calypso web (same site) on iPhone SE:
![iosreferrers](https://user-images.githubusercontent.com/15107387/81057561-f4d08400-8ecc-11ea-91cc-79f7825481a4.png)

##### Tested on iPhone SE, iOS 13.1.2, WPiOS 14.6
This came up in #2934822-zen when a user wanted to mark some referrers as spam but couldn't.

**Comment by thehenrybyrd** [comment]
Same issue in Android: https://github.com/wordpress-mobile/WordPress-Android/issues/11818

**Comment by leandroalonso** [comment]
@SylvesterWilmott any input on how we can implement a button to access this feature on iOS? I think we can add the "..." there but the cells already have the chevron, so not sure. Long press (the solution on the Android side) is something that I don't think is that common on iOS.

(Another option might be sliding the cell, it's still hidden but it's more common.)

**Comment by SylvesterWilmott** [comment]
@leandroalonso I think the ellipsis icon should work. Currently, the rows with the "**>**" chevron do not trigger a push animation, so those chevrons are redundant and can be replaced with the ellipsis to trigger an action sheet. To indicate that the referrer rows do something when tapped, blue text can be used for the rows that open a modal. See below for illustration.

<img width="417" alt="Screenshot 2021-06-03 at 12 55 30" src="https://user-images.githubusercontent.com/36532468/120641001-06d37c00-c46b-11eb-8b61-549e1bddd392.png">

**Comment by nikola-milicevic** [comment]
Hey @SylvesterWilmott, thanks for jumping in!
Just a question, should referrers marked as spam have a different look?

**Comment by SylvesterWilmott** [comment]
@leandroalonso correct me if I am wrong, but referrers marked as spam should be removed from the list. On web there is a window of opportunity to undo marking as spam, would we typically use a snackbar in this scenario? I'm a little out of the WP loop. Maybe @osullivanchris can help us out.

**Comment by leandroalonso** [comment]
@SylvesterWilmott yes, they're removed. When you mark it as spam you can undo the operation (and the row is still there), however, when you load it again it won't be there.

I think we can follow the same Android behavior:

- When marked as spam the row greys out
- If the user tap "..." again the option would be to "Unmark as spam"
- If the user unmarks it, then the row is displayed normally again

In the subsequent times, the row is simply not displayed.

**Comment by SylvesterWilmott** [comment]
Sounds good to me, thanks @leandroalonso

**Comment by nikola-milicevic** [comment]
Hey @SylvesterWilmott 👋

@emilylaguna and I need your help again 😊
I went through the web version, and I think the [design you suggested](https://github.com/wordpress-mobile/WordPress-iOS/issues/14055#issuecomment-853812899) might need to be tweaked a bit.

In web version you can mark the 'parent' referrer as spam, and not the children.

So parent has two potential actions:
- expanding / collapsing
- marking as spam

And child only one:
- opening the browser with that child URL

My proposal for parent cell is:
- make the text blue
- leave the chevron on the right
- tapping on a cell (that's indicated by blue text) would open a sheet with 'Mark' / 'Unmark as spam' action
- tapping on a chevron would do expand / collapse

Child cell will have only disclosure indicator on the right, and tapping anywhere inside the cell would open the browser.

Hope I was clear, let me know what you think! 😊
